### PR TITLE
Add cached libvirt

### PR DESCRIPTION
**Pull request by aarondl**

### CODE REVIEW

File: packer/http/autoyast.xml.tpl [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+  <language>
+    <language>en_US</language>
+    <languages>en_US</languages>
+  </language>
+
+  <networking>
+    <dhcp_options>
+      <dhclient_client_id/>
+      <dhclient_hostname_option>AUTO</dhclient_hostname_option>
+    </dhcp_options>
```
Comment (mook-as) [inline packer/http/autoyast.xml.tpl]: I know you didn't change this, but I wonder if we could drop the whole `<dhcp_options>` block?

Not worth a rebuild to test, though

File: packer/http/scfyast.xml [hunk 2]
```diff
@@ -35,7 +35,7 @@
     </dhcp_options>
 
     <dns>
-      <hostname>vagrant-kube</hostname>
+      <hostname>scf-kube</hostname>
```
Comment (mook-as) [inline packer/http/scfyast.xml]: Shouldn't this file no longer exist?

File: packer/scripts/install-helm-images.rb [hunk 3]
```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env ruby
```
Comment (mook-as) [inline packer/scripts/install-helm-images.rb]: Is this file still used?

File: packer/http/scfyast.xml (hunk 2 above)
Comment (aarondl) [reply packer/http/scfyast.xml]: Shouldn't deleted.

File: packer/scripts/install-helm-images.rb (hunk 3 above)
Comment (aarondl) [reply packer/scripts/install-helm-images.rb]: Shouldn't deleted.

File: packer/README [hunk 4]
```diff
@@ -1,6 +1,14 @@
...
+packer: https://www.packer.io/
+txtplate: https://github.com/aarondl/txtplate/releases
+
+## Creation
+
+To create one of the boxes do this
+
 ```bash
-packer build -only=<ONE OF: virtualbox-iso, qemu, vmware-iso> vagrant-box.json
+packer build -only=<ONE OF: vagrant-virtualbox, vagrant-libvirt, vagrant-vmware, jenkins-slave, scf-cached> vagrant-box.json
```
[the remark this answers is not in the document]
Comment (aarondl) [reply packer/README]: Yes :D Fixed.

File: packer/Makefile [hunk 5]
```diff
@@ -0,0 +1,19 @@
+.PHONY: vagrant-virtualbox vagrant-libvirt scf-cached
+
+scf-cached-libvirt: http/scf-autoyast.xml
+	packer build -only scf-cached-libvirt vagrant-box.json
+vagrant-libvirt: http/vagrant-autoyast.xml
+	packer build -only vagrant-libvirt vagrant-box.json
+vagrant-virtualbox: http/vagrant-autoyast.xml
+	packer build -only vagrant-virtualbox vagrant-box.json
+
+# The following build pieces require txtplate to evaluate Go templates
```
[the remark this answers is not in the document]
Comment (aarondl) [reply packer/Makefile]: This doesn't take away enough duplication to be worth the complexity in the statement. Can't bring myself to do this to a later reader.

File: packer/Makefile [hunk 6]
```diff
@@ -0,0 +1,19 @@
+.PHONY: vagrant-virtualbox vagrant-libvirt scf-cached
```
[the remark this answers is not in the document]
Comment (aarondl) [reply packer/Makefile]: Sounds good to me. Fixed.

File: packer/Makefile (hunk 5 above)
[the remark this answers is not in the document]
Comment (mook-as) [reply packer/Makefile]: At least make the indenting in the two rules line up, then?

File: packer/README [hunk 7]
```diff
@@ -1,6 +1,14 @@
...
+packer: https://www.packer.io/
+txtplate: https://github.com/aarondl/txtplate/releases
+
+## Creation
+
+To create one of the boxes do this
+
 ```bash
-packer build -only=<ONE OF: virtualbox-iso, qemu, vmware-iso> vagrant-box.json
+make <ONE OF: vagrant-virtualbox, vagrant-libvirt, vagrant-vmware, jenkins-slave, scf-cached>
```
Comment (mook-as) [inline packer/README]: You don't actually have `make` rules for `vagrant-vmware` or `jenkins-slave`; and the last one actually has a target of `scf-cached-libvirt`, not `scf-cached`.

File: packer/Makefile [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
```
Comment (mook-as) [inline packer/Makefile]: Hmm, `scf-cached` isn't a valid target here. (It's being hit by my tab completion)

Comment (aarondl) [reply packer/Makefile]: Fixed.

**Comment by mook-as** [comment]
Sorry, your cached VM is now busted because you need the [RBAC chunk](https://github.com/SUSE/scf/blob/e31c7ffe41fce801a633a83898f6c602afdb4e1d/bin/common/install_tools.sh#L51-L69) in [install-helm.sh](https://github.com/SUSE/scf/blob/481a60a91be9b90b8ac4fed8497671291f148578/packer/scripts/install-helm.sh).

**Pull request merged**